libsolidity/interface/CompilerStack.cpp

```diff
@@ -1477,15 +1477,8 @@ void CompilerStack::compileContract(
 	solAssert(!m_viaIR, "");
 	bytes cborEncodedMetadata = createCBORMetadata(compiledContract, /* _forIR */ false);
 
-	try
-	{
-		// Run optimiser and compile the contract.
-		compiler->compileContract(_contract, _otherCompilers, cborEncodedMetadata);
-	}
-	catch(evmasm::OptimizerException const&)
-	{
-		solAssert(false, "Optimizer exception during compilation");
-	}
+	// Run optimiser and compile the contract.
+	compiler->compileContract(_contract, _otherCompilers, cborEncodedMetadata);
```

Comment:
I ran into this when I forgot to update `GasMeter::runGas()` and got an ICE without the original message and pointing here instead of the place where the problem happened. Since `OptimizerException` inherits from `util::Exception` and we have top-level handlers for that, I think it's better to just let it through.